### DISCUSSION TITLE
What can be used instead of  dfs.find_nearest_profile_elements(long, lat)

### DISCUSSION THREAD

**Discussion by bhlevca** [DHI/mikeio#596] [Q&A] [answered]
Hello,

DFSU's` find_nearest_profile_elements(long, lat)` seems to have been removed.

I used to get the element ids for the long/lat combination like this 
 `elem_ids = dfs.find_nearest_profile_elements(x, y)`

What would be the best option to achieve the same result with the current version (1.6.2)

Thanks
Bogdan

**Comment by ecomodeller** [comment, marked as answer]
How about `dfs.read(x, y)` ?
![image](https://github.com/DHI/mikeio/assets/614215/1cc684f5-fc59-4721-924f-1937f5746d58)

**Reply by bhlevca** [reply]
Thanks, did the trick!

**Reply by bhlevca** [reply]
But I noticed something strange.
read(x,y) takes approximately 20 times longer on Linux than on Windows. The Linux machine was a powerful workstation and the Windows machine was a laptop.  That means that the difference is even bigger.  
I am going to file a bug report. Python 3.11.4 on both. mikeio 1.6.2 on both.

**Reply by ecomodeller** [reply]
We are considering to extend the subsetting options for 3d files, @bhlevca do you have any suggestions on useful features that would ease the work with Dfsu 3d data?

**Comment by bhlevca** [comment]
Let me think about it.  However, from the top of my mind, I often work with
field data and I need to drop a lat/long when I have the instruments and
get the time series for that location out of the model file.
dfs.read(x, y) kind of does that, but dealing with the vertical profiles is
cumbersome. It can be done, but having an explicit API that will allow
getting that data directly for a location and depth would simplify and
speed up the code

How about read(x, y, z) ?

I'll think about other uses after our Thanksgiving long weekend ;-)

Thanks
Bogdan

**Reply by ecomodeller** [reply]
No rush, we will start this development in 2024 (if prioritized).
